pennylane/transforms/optimization/merge_rotations.py

```diff
@@ -25,7 +28,211 @@
+
+            # Refresh the previous_ops dictionary with the current operator
+            for o in previous_ops_on_wires:
+                if o is not None:
+                    for w in o.wires:
+                        del self.previous_ops[w]
+            for w in op.wires:
+                self.previous_ops[w] = op
+
+            results = []
+            for prev_op in previous_ops_on_wires:
+                results.append(super().interpret_operation(prev_op))
+            return results
+
+        # pylint: disable=inconsistent-return-statements
+        def interpret_operation(self, op: Operator):
+            """Interpret a PennyLane operation instance.
+
+            Args:
+                op (Operator): a pennylane operator instance
+
+
+            This method is only called when the operator's output is a dropped variable,
+            so the output will not affect later equations in the circuit.
+
+            See also: :meth:`~.interpret_operation_eqn`.
+            """
+
+            if self.include_gates is not None and op.name not in self.include_gates:
+                self.interpret_and_refresh_previous_ops(op)
+                return
+
+            if op not in composable_rotations:
+                self.interpret_and_refresh_previous_ops(op)
+                return
+
+            previous_op = self.previous_ops.get(op.wires[0])
+            if previous_op is None:
+                for w in op.wires:
+                    self.previous_ops[w] = op
+                return
+
+            wires_exactly_match = op.wires == previous_op.wires
+            are_same_type = isinstance(op, type(previous_op))
+            can_merge = wires_exactly_match and are_same_type
+
+            if not can_merge:
+                self.interpret_and_refresh_previous_ops(op)
+                return
+
+            if isinstance(op, qml.Rot):
+                # Order of arguments matter for the Rot gate!
+                cumulative_angles = fuse_rot_angles(
+                    qml.math.stack(previous_op.parameters),
+                    qml.math.stack(op.parameters),
+                )
+                test_angles = qml.math.stack(
+                    [cumulative_angles[0] + cumulative_angles[2], cumulative_angles[1]]
+                )
+            else:
+                cumulative_angles = qml.math.stack(previous_op.parameters) + qml.math.stack(
+                    op.parameters
+                )
+                test_angles = cumulative_angles
+
+            angles_cancel = qml.math.allclose(test_angles, 0.0, atol=self.atol, rtol=0)
+            keep_merged_op = (
+                qml.math.is_abstract(cumulative_angles)
+                or qml.math.requires_grad(cumulative_angles)
+                or not angles_cancel
+            )
+            if keep_merged_op:
+                # pylint: disable = protected-access
+                new_op = op.__class__._primitive.impl(*cumulative_angles, wires=op.wires)
```

Comment:
Thoughts?
```suggestion
                with qml.capture.pause():
                    new_op = type(op)(*cumulative_angles, wires=op.wires)
```

Reply:
We also don't need to use `__class__` here if you prefer to not use the above suggestion
```suggestion
                new_op = op._primitive.impl(*cumulative_angles, wires=op.wires)
```